src/lib/stores/orderStore.ts

```diff
@@ -0,0 +1,102 @@
+import { PUBLIC_PB_HOST } from "$env/static/public";
+import { writable } from "svelte/store";
+import pocketbase from "pocketbase";
+import { Order } from "$lib/types";
+import type { State } from "$lib/types";
+
+// ref: https://github.com/pocketbase/js-sdk?tab=readme-ov-file#nodejs-via-npm
+import eventsource from "eventsource";
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+(global as any).EventSource = eventsource;
+
+const pb = new pocketbase(PUBLIC_PB_HOST);
+
+const mapToOrder = (data: unknown): Order => {
+  return new Order({
+    id: data.id,
+    state: data.state,
+    collectionId: data.collectionId,
+    collectionName: data.collectionName,
+    created: data.created,
+    customer: data.customer,
+    drinks: data.drinks,
+    payment_fulfilled: data.payment_fulfilled,
+    updated: data.updated
+  });
+};
+
+const init = () => {
+  const { subscribe, set, update } = writable<Order[]>([]);
+
+  (async () => {
+    const initialOrders = await pb.collection("orders").getFullList();
+    set(initialOrders.map(mapToOrder));
+
+    pb.collection("orders").subscribe("*", (e) => {
+      update((state) => {
+        console.log({
+          message: "received event on orders subscription",
+          event: e,
+          currentState: state
+        });
+
+        const orderIndex = state.findIndex((order) => order.id === e.record.id);
+
+        const order = mapToOrder(e.record);
+
+        switch (e.action) {
+          case "create":
+            state.push(order);
+            break;
+          case "update":
+            state[orderIndex] = order;
+            break;
+          case "delete":
+            state.splice(orderIndex, 1);
+            break;
+        }
+
+        return state;
+      });
+    });
+  })();
+
+  return {
+    subscribe,
+    add: async (order: string[]) => {
+      const drinkIds: string[] = await Promise.all(
+        // this mapping to create multiple order_drink with autocancellation disabled
+        // equals a terrible idea
+        // bulk / transactions coming in 0.23
+        // https://github.com/pocketbase/pocketbase/issues/5386
+        order.map(async (id: string) => {
+          const response = await pb.collection("order_drink").create(
+            {
+              drink: id,
+              serving_size: "big",
```

Comment:
Kanskje vente med å implementere slikt til senere? Trengs ikke med en gang for MVP

Reply:
Mener du `serving_size`, etc? Jeg tenker dette er et funksjonelt krav, og det å tenke på størrelsen av koppen, pluss pluss, er en normal dag på kaffe-diem. Da er det greit at vi kan representere det.

Reply:
jeg har avventet med å vise noe i denne PR. Det er fullt mulig å ha MVP til opprinnelig demo. Kaffe-diem kommer unngaelig til å ha noen forslag, og da er det relevant til å nevne dette